DSpace/DSpace
# Refactor/remove custom MultipartFileSender to use Spring Boot Range requests

## References
* Fixes #2983 
* Fixes https://github.com/DSpace/dspace-angular/pull/827

## Description
The custom [MultipartFileSender](https://github.com/DSpace/DSpace/blob/main/dspace-server-webapp/src/main/java/org/dspace/app/rest/utils/MultipartFileSender.java) was completely bypassing Spring Security & Spring automated tools.
This PR makes sure that we lean more towards the 'default' usages of the previously mentioned Spring Boot etc.

## Instructions for Reviewers
Please add a more detailed description of the changes made by your PR. At a minimum, providing a bulleted list of changes in your PR is helpful to reviewers.

### List of changes in this PR:
* MultipartFileSender is renamed to HttpHeadersInitializer to more correctly reflect the functionality/usage
* HttpHeadersInitializer been 'changed' to now only initialise the HTTPHeaders, apart from trying to handle a whole bunch of code itself (So that we can rely on the spring framework to handle the MultiPart file sending)
    * The HTTPHeaders will be used in the ResponseEntity given by any methods that need to reply with a downloaded file
* The SitemapRestController is altered to adhere to the guidelines of returning a ResponseEntity as well
* the BitstreamRestController returns a new ResponseEntity for the bitstream resource to properly handle the multipart file sending

### Testing steps using REST
* Setup the related REST code
* Trigger a script that keeps output in a file
* Trigger a GET call to that particular output bitstream (With an origin header)
* Check that the header "Access-Control-Allow-Origin:" is returned as expected
### Testing steps using Angular
* Setup the related REST code
* Setup the [following Angular PR](https://github.com/DSpace/dspace-angular/pull/827)
* Make sure to trigger a script that keeps output in a file.
* Try to download that particular output bitstream
* The bitstream should be properly 'downloaded', no CORS error should be logged, and the information of the logs should be shown in the UI

## Checklist
_This checklist provides a reminder of what we are going to look for when reviewing your PR. You need not complete this checklist prior to creating your PR (draft PRs are always welcome). If you are unsure about an item in the checklist, don't hesitate to ask. We're here to help!_
